core/common/src/main/java/alluxio/underfs/BaseUnderFileSystem.java

```diff
@@ -86,6 +89,23 @@ public String getFingerprint(String path) {
     }
   }
 
+  @Override
+  public UfsMode getOperationMode(Map<String, UfsMode> physicalUfsState) {
+    for (Map.Entry<String, UfsMode> entry : physicalUfsState.entrySet()) {
+      if (UnderFileSystemUtils.stripFolderFromPath(entry.getKey())
+          .equals(UnderFileSystemUtils.stripFolderFromPath(mUri.toString()))) {
+        return entry.getValue();
+      }
+    }
+    return UfsMode.READ_WRITE;
+  }
+
+  @Override
+  public List<String> getPhysicalUfs() {
```

Comment:
I was struggling to come up with a good name. "Physical" vs "logical" stemmed from RAID terminology of multiple physical disks making up a logical disk. I can add the root path to `AlluxioUri` as you suggested, but keep the term physical ufs?. @gpang @calvinjia what do you think?.

Reply:
`getPhysicalStores`? The name does not seem ambiguous to me because nothing mentions `path` here, so I assume only scheme and authority. Alternatively we could return a non-String object which would make it type safe.